{
  "cve_id": "CVE-2022-1319",
  "github_data": {
    "repository": "undertow-io/undertow",
    "fix_commit": "1443a1a2bbb8e32e56788109d8285db250d55c8b",
    "related_commits": [
      "1443a1a2bbb8e32e56788109d8285db250d55c8b",
      "7c5b3ab885b5638fd3f1e8a935d5063d68aa2df3",
      "1443a1a2bbb8e32e56788109d8285db250d55c8b",
      "7c5b3ab885b5638fd3f1e8a935d5063d68aa2df3"
    ],
    "patch_url": "https://github.com/undertow-io/undertow/commit/1443a1a2bbb8e32e56788109d8285db250d55c8b.patch",
    "fix_commit_details": {
      "sha": "1443a1a2bbb8e32e56788109d8285db250d55c8b",
      "commit_date": "2022-04-08T21:18:29Z",
      "author": {
        "login": "aogburn",
        "type": "User",
        "stats": null
      },
      "commit_message": {
        "title": "[UNDERTOW-2060] fix double AJP response",
        "length": 39,
        "has_description": false,
        "references_issue": true
      },
      "stats": {
        "total": 3,
        "additions": 1,
        "deletions": 2
      },
      "files": [
        {
          "filename": "core/src/main/java/io/undertow/server/protocol/ajp/AjpReadListener.java",
          "status": "modified",
          "additions": 1,
          "deletions": 2,
          "patch": "@@ -58,7 +58,7 @@ final class AjpReadListener implements ChannelListener<StreamSourceChannel> {\n     private static final byte[] CPONG = {'A', 'B', 0, 1, 9};\n     private static final byte[] SEND_HEADERS_INTERNAL_SERVER_ERROR_MSG = {'A', 'B', 0, 8, 4, (byte)((500 >> 8) & 0xFF) , (byte)(500 & 0xFF), 0, 0, '\\0', 0, 0};\n     private static final byte[] SEND_HEADERS_BAD_REQUEST_MSG = {'A', 'B', 0, 8, 4, (byte)((400 >> 8) & 0xFF) , (byte)(400 & 0xFF), 0, 0, '\\0', 0, 0};\n-    private static final byte[] END_RESPONSE = {'A', 'B', 0, 2, 5, 1};\n+    private static final byte[] END_RESPONSE = {'A', 'B', 0, 2, 5, 0};\n \n     private final AjpServerConnection connection;\n     private final String scheme;\n@@ -235,7 +235,6 @@ public void handleEvent(AjpServerResponseConduit channel) {\n             if(oldState.badRequest) {\n                 httpServerExchange.setStatusCode(StatusCodes.BAD_REQUEST);\n                 httpServerExchange.endExchange();\n-                handleBadRequest();\n                 safeClose(connection);\n             } else {\n                 Connectors.executeRootHandler(connection.getRootHandler(), httpServerExchange);"
        }
      ],
      "file_patterns": {
        "security_files": 0,
        "config_files": 0,
        "dependency_files": 0,
        "test_files": 0,
        "unique_directories": 1,
        "max_directory_depth": 9
      },
      "context": {
        "surrounding_commits": [
          {
            "sha": "6ae61c6af88d2a8341922ccd0de98926e8349543",
            "date": "2024-12-06T13:14:17Z",
            "author_login": "baranowb"
          },
          {
            "sha": "e9c4ed891ba910e557ce8ee1491d57635cfe6a22",
            "date": "2024-12-04T11:29:37Z",
            "author_login": "ropalka"
          },
          {
            "sha": "84c2d6ca66c7a4882f4de212a621c0682bbe8635",
            "date": "2024-12-04T09:33:21Z",
            "author_login": "fl4via"
          },
          {
            "sha": "a4720e31097e51d1dc72596ccbb9e0e7c90e3bf3",
            "date": "2024-11-28T08:56:47Z",
            "author_login": "baranowb"
          },
          {
            "sha": "29fc22ca79cf4f0a96c0e9d141f418a25455e8cb",
            "date": "2024-11-27T07:41:41Z",
            "author_login": "baranowb"
          }
        ]
      }
    }
  },
  "vulnerability_details": {
    "cvss_score": 7.5,
    "cvss_vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
    "cwe_id": "CWE-252",
    "description": "A flaw was found in Undertow. For an AJP 400 response, EAP 7 is improperly sending two response packets, and those packets have the reuse flag set even though JBoss EAP closes the connection. A failure occurs when the connection is reused after a 400 by CPING since it reads in the second SEND_HEADERS response packet instead of a CPONG.",
    "attack_vector": "NETWORK",
    "attack_complexity": "LOW"
  },
  "temporal_data": {
    "published_date": "2022-08-31T16:15:09.410",
    "last_modified": "2024-11-21T06:40:28.920",
    "fix_date": "2022-04-08T21:18:29Z"
  },
  "references": [
    {
      "url": "https://access.redhat.com/security/cve/CVE-2022-1319",
      "source": "secalert@redhat.com",
      "tags": [
        "Vendor Advisory"
      ]
    },
    {
      "url": "https://bugzilla.redhat.com/show_bug.cgi?id=2073890",
      "source": "secalert@redhat.com",
      "tags": [
        "Issue Tracking",
        "Vendor Advisory"
      ]
    },
    {
      "url": "https://github.com/undertow-io/undertow/commit/1443a1a2bbb8e32e56788109d8285db250d55c8b",
      "source": "secalert@redhat.com",
      "tags": [
        "Patch",
        "Third Party Advisory"
      ]
    },
    {
      "url": "https://github.com/undertow-io/undertow/commit/7c5b3ab885b5638fd3f1e8a935d5063d68aa2df3",
      "source": "secalert@redhat.com",
      "tags": [
        "Patch",
        "Third Party Advisory"
      ]
    },
    {
      "url": "https://issues.redhat.com/browse/UNDERTOW-2060",
      "source": "secalert@redhat.com",
      "tags": [
        "Issue Tracking",
        "Vendor Advisory"
      ]
    },
    {
      "url": "https://security.netapp.com/advisory/ntap-20221014-0006/",
      "source": "secalert@redhat.com",
      "tags": [
        "Third Party Advisory"
      ]
    },
    {
      "url": "https://access.redhat.com/security/cve/CVE-2022-1319",
      "source": "af854a3a-2127-422b-91ae-364da2661108",
      "tags": [
        "Vendor Advisory"
      ]
    },
    {
      "url": "https://bugzilla.redhat.com/show_bug.cgi?id=2073890",
      "source": "af854a3a-2127-422b-91ae-364da2661108",
      "tags": [
        "Issue Tracking",
        "Vendor Advisory"
      ]
    },
    {
      "url": "https://github.com/undertow-io/undertow/commit/1443a1a2bbb8e32e56788109d8285db250d55c8b",
      "source": "af854a3a-2127-422b-91ae-364da2661108",
      "tags": [
        "Patch",
        "Third Party Advisory"
      ]
    },
    {
      "url": "https://github.com/undertow-io/undertow/commit/7c5b3ab885b5638fd3f1e8a935d5063d68aa2df3",
      "source": "af854a3a-2127-422b-91ae-364da2661108",
      "tags": [
        "Patch",
        "Third Party Advisory"
      ]
    },
    {
      "url": "https://issues.redhat.com/browse/UNDERTOW-2060",
      "source": "af854a3a-2127-422b-91ae-364da2661108",
      "tags": [
        "Issue Tracking",
        "Vendor Advisory"
      ]
    },
    {
      "url": "https://security.netapp.com/advisory/ntap-20221014-0006/",
      "source": "af854a3a-2127-422b-91ae-364da2661108",
      "tags": [
        "Third Party Advisory"
      ]
    }
  ],
  "collection_metadata": {
    "collected_at": "2025-01-11T23:03:39.063006",
    "processing_status": "raw"
  },
  "repository_context": {
    "name": "undertow",
    "owner": "undertow-io",
    "created_at": "2012-07-19T00:52:44Z",
    "updated_at": "2025-01-06T03:33:15Z",
    "pushed_at": "2024-12-06T13:14:17Z",
    "size": 27427,
    "stars": 3615,
    "forks": 1009,
    "open_issues": 64,
    "watchers": 3615,
    "has_security_policy": false,
    "default_branch": "main",
    "protected_branches": [
      "1.0.x",
      "1.1.x",
      "1.2.x",
      "1.3.x",
      "1.4.x",
      "1.4.18.x",
      "main"
    ],
    "languages": {
      "Java": 8854425,
      "HTML": 1782,
      "Shell": 86
    },
    "commit_activity": {
      "total_commits_last_year": 0,
      "avg_commits_per_week": 0,
      "days_active_last_year": 0
    },
    "security_features": {
      "has_security_policy": false,
      "has_protected_branches": true,
      "has_wiki": false,
      "has_issues": false,
      "allow_forking": true,
      "is_template": false,
      "license": "apache-2.0"
    },
    "collected_at": "2025-01-14T14:58:25.169097"
  }
}